toolbox/src/base/Signal.cpp

```diff
@@ -223,19 +277,19 @@ namespace wbt {
             case PortDataTypeUInt32:
             {
                 dataSize = sizeof(uint32_t);
-                address = ((uint32_t*)data) + startIndex;
+                address = data + startIndex;
                 break;
             }
             case PortDataTypeUInt16:
             {
                 dataSize = sizeof(uint16_t);
-                address = ((uint16_t*)data) + startIndex;
+                address = data + startIndex;
                 break;
             }
             case PortDataTypeUInt8:
             {
                 dataSize = sizeof(uint8_t);
-                address = ((uint8_t*)data) + startIndex;
+                address = data + startIndex;
```

Comment:
Why did you drop the cast?

Reply:
Replied [here](https://github.com/robotology/WB-Toolbox/pull/56#discussion_r150213326)